controllers/rbac/aggregation.go

```diff
@@ -0,0 +1,262 @@
+	return default_namespace
+}
+
+// NewAggregation create a new Aggregation struct
+func NewAggregation(mgr ctrl.Manager, namespace string) Aggregation {
+	return &aggregation{
+		Client:    mgr.GetClient(),
+		reader:    mgr.GetAPIReader(),
+		namespace: namespace,
+	}
+}
+
+func (a aggregation) CreateOrUpdateAggregation() error {
+	fmt.Println("Got to CreateOrUpdateAggregation")
+	if err := a.createOrUpdateRole(); err != nil {
+		return err
+	}
+	return a.createOrUpdateRoleBinding()
+
+}
+
+func (a aggregation) createOrUpdateRole() error {
+	// check if the role exists
+	fmt.Println("Got to createOrUpdateRole")
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: default_namespace,
+			// OwnerReferences: getOwnerRefs(reader, namespace),
+		},
+		// AggregationRule: &rbacv1.AggregationRule{
+		// 	ClusterRoleSelectors: []metav1.LabelSelector{
+		// 		{
+		// 			MatchLabels: map[string]string{
+		// 				aggregationLabelKey: "true",
+		// 			},
+		// 		},
+		// 	},
+		// },
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"*"}, // "get",
+				// "list",
+				// "watch",
+				// "create",
+				// "update",
+				// "patch",
+				// "delete",
+			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"*"}, // "get",
+				// "list",
+				// "watch",
+				// "create",
+				// "update",
+				// "patch",
+				// "delete",
+
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events"},
+				Verbs: []string{
+					"create",
+					"patch",
+				},
+			},
+		},
+	}
+	err := a.reader.Get(context.Background(), client.ObjectKeyFromObject(role), role)
+	if errors.IsNotFound(err) {
+		return a.createRole()
+	} else if err != nil {
+		return fmt.Errorf("failed to get cluster role: %v", err)
+	}
+	return a.updateRole(role)
+}
+func (a aggregation) createRole() error {
+	err := a.Create(context.Background(), getRole())
+	return err
+}
+
+func (a aggregation) updateRole(oldRole *rbacv1.Role) error {
+	newRole := getRole()
+	oldRole.Rules = newRole.Rules
+	// oldRole.AggregationRule = newRole.AggregationRule
+	return a.Update(context.Background(), oldRole)
+}
+
+// func getRole(reader client.Reader, namespace string) *rbacv1.ClusterRole {
+func getRole() *rbacv1.Role {
+	fmt.Println("Got to getRole")
+	return &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: default_namespace,
+			// OwnerReferences: getOwnerRefs(reader, namespace),
+		},
+		// AggregationRule: &rbacv1.AggregationRule{
+		// 	ClusterRoleSelectors: []metav1.LabelSelector{
+		// 		{
+		// 			MatchLabels: map[string]string{
+		// 				aggregationLabelKey: "true",
+		// 			},
+		// 		},
+		// 	},
+		// },
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"*"}, // "get",
```

Comment:
and to make it worse: you need to change leader election manifests to clusterrole / -binding, because otherwise we can't create roles for the default namespace (we can't create more specific role / binding because we don't know where our SA is located... which closes the circle)